### PR TITLE
make sure dns name is set correctly for kubernetes services

### PR DESCRIPTION
**Pull request by wlan0**
@alena1108

### CODE REVIEW

**Comment by wlan0** [comment]
Is required to fix - https://github.com/rancher/rancher/issues/5236

**Comment by wlan0** [comment]
This PR makes the kubernetes service adhere to the format defined under "network identity" subsection of these docs - http://kubernetes.io/docs/user-guide/petset/

File: code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/DnsInfoDaoImpl.java [hunk 1]
```diff
@@ -360,11 +360,17 @@ protected void addToResolve(ServiceInstanceData targetInstance, String aliasName
                 if (StringUtils.isEmpty(targetInstanceName)) {
                     ipToInstanceName.put(targetIp, null);
                 } else {
+                    String dnsName;
+                    if (targetInstance.getService().getKind().equalsIgnoreCase("kubernetesservice")) {
```
Comment (ibuildthecloud) [inline code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/DnsInfoDaoImpl.java]: @alena1108 Can we have an approach that doesn't hard code logic to k8s?

Comment (alena1108) [inline code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/DnsInfoDaoImpl.java]: @ibuildthecloud as this split just changes fqdn format from instanceName.globalNamespace (followed by regular service instances) to instanceName.serviceName.stackName.globalNamespace (followed by k8s service instances), so we can go with the label approach. The label would define whether svcName.stackName should be included to the individual container resolution.

**Pull request merged**